### PR TITLE
WIP: Create builds from config

### PR DESCRIPTION
**Pull request by LindsayYoung**
This should be a deployment improvement. We can configure the commands here, so we don't have to run separate drush commands. 

Thanks Marvin!

I committed the existing file but only made 2 small changes. I haven't tested it out yet.

### CODE REVIEW

**Comment by alexsoble** [comment]
@LindsayYoung Hmm, the `/vendor` file is gitignored, correct? Since that's the case, how are we committing this file?

**Comment by LindsayYoung** [comment]
Yes, we only want to commit this file with the config.

File: vendor/acquia/blt/config/build.yml [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    # E.g., $config_directories['sync']. It must have a corresponding key in cm.core.dirs. E.g., `cm.core.dirs.sync`.
+    key: sync
+    dirs:
+      # Corresponding value is defined in config.settings.php.
+      sync:
+        path: ${cm.core.path}/default
+    # Install site directly from existing config.
+    # This cannot be used if your install profile implements hook_install.
+    # Import our project config
+    install_from_config: true
```
Comment (LindsayYoung) [inline vendor/acquia/blt/config/build.yml]: added this to import our project config

**Comment by alexsoble** [comment]
@LindsayYoung Ah, I see ...

```
git add --force my/ignore/file.foo
```

https://stackoverflow.com/questions/8006393/force-add-despite-the-gitignore-file

File: vendor/acquia/blt/config/build.yml (hunk 1 above)
Comment (alexsoble) [reply vendor/acquia/blt/config/build.yml]: @LindsayYoung Looking at what this does in the BLT code ... 👀 

https://github.com/acquia/blt/blob/46203b6c9c55ef142c9bd9150c7e6d2a14bfd5a0/src/Robo/Commands/Setup/DrupalCommand.php#L61

**Comment by LindsayYoung** [comment]
Tried this on Friday morning and it did not deploy the config. We are probably missing a config the one we set is only for initializing drupal

File: vendor/acquia/blt/config/build.yml [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+  strategy: install
+  # If setup.strategy is import, this file will be imported. File path is
+  # relative to drupal docroot directory.
+  dump-file: null
+
+sync:
+  # By default, files are not synced during sync:refresh.
+  # Set this value to 'true' or pass -D sync.files=true
+  # to override this behavior.
+  files: false
```
Comment (LindsayYoung) [inline vendor/acquia/blt/config/build.yml]: I think this needs to be true too, I am going to try it.

Comment (alexsoble) [reply vendor/acquia/blt/config/build.yml]: Ooh, good idea!

File: vendor/acquia/blt/config/build.yml [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+  strategy: install
+  # If setup.strategy is import, this file will be imported. File path is
+  # relative to drupal docroot directory.
+  dump-file: null
+
+sync:
+  # By default, files are not synced during sync:refresh.
+  # Set this value to 'true' or pass -D sync.files=true
+  # to override this behavior.
+  files: true
```
Comment (LindsayYoung) [inline vendor/acquia/blt/config/build.yml]: Added this too, though looks like we are missing something, to get to the point where we don't also need to run the drush commands.

Comment (alexsoble) [reply vendor/acquia/blt/config/build.yml]: Hmmmmm, let's check in with Marvin from Acquia on Monday, he should know!

Comment (alexsoble) [reply vendor/acquia/blt/config/build.yml]: I looked to see if there was anything we could borrow from https://github.com/usdoj/foia-api but that repo doesn't have a `vendor` directory so they must not have gone the force-git-adding route.

**Comment by LindsayYoung** [comment]
Did get the site name to update on staging but not dev, I think there might be more settings before we don't need the drush commands. We can check with Marvin on Monday

File: vendor/acquia/blt/config/build.yml [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    - files.php.custom.themes
+    - files.frontend.custom.themes
+
+project:
+  local:
+    uri: ${project.local.protocol}://${project.local.hostname}
+
+setup:
+  # Valid values are install, sync, import.
+  strategy: install
```
Comment (LindsayYoung) [inline vendor/acquia/blt/config/build.yml]: change to sync

**Comment by LindsayYoung** [comment]
closing this in favor of #16

**Pull request closed without merging**